src/tests/class_tests/openms/source/MRMScoring_test.cpp

```diff
@@ -288,15 +297,20 @@ mean(xcorr_max) # shape score
           TEST_EQUAL(mrmscore.getXCorrPrecursorCombinedMatrix().cols(), 5)
 
           std::vector<double> sum_matrix;
-          for(auto e : mrmscore.getXCorrPrecursorCombinedMatrix())
-          {
-            double sum{0};
-            for(size_t i = 0; i < e.data.size(); ++i)
+
+          const auto& cm = mrmscore.getXCorrPrecursorCombinedMatrix();
```

Comment:
same here:
this "reduction" code that has vectors of data *inside* matrix cells fill the sum_matrix in the order of row/col traversal.